### PR TITLE
E2E: extend the "inject" test with prefix lists

### PR DESCRIPTION
**Pull request by fedepaol**
Users requiring routes to come from outside may need to filter only allowed routes. Here we validate that it works.

### CODE REVIEW

File: e2etest/bgptests/bgp.go [hunk 1]
```diff
@@ -1108,11 +1113,17 @@ var _ = ginkgo.Describe("BGP", func() {
 					if err == nil {
 						return fmt.Errorf("route not injected from %s", c.Name)
 					}
+					err = checkRouteInjected(speakerPods, pairingIPFamily, toFilter, c.RouterConfig.VRF)
```
Comment (mlguerrero12) [inline e2etest/bgptests/bgp.go:1116]: not in the scope of this PR but I'd change the name to isRouteInjected and the return value to be a boolean. Based on that you can return an error in the closure. It's very misleading the way it is (return error when route is injected).

**Comment by mlguerrero12** [comment]
lgtm

File: e2etest/bgptests/bgp.go [hunk 2]
```diff
@@ -1108,11 +1113,17 @@ var _ = ginkgo.Describe("BGP", func() {
...
+					if err != nil {
+						return fmt.Errorf("to filter route injected from %s, %w", c.Name, err)
```
Comment (liornoy) [inline e2etest/bgptests/bgp.go:1118]: tiny nit
```suggestion
						return fmt.Errorf("failed to filter route injected from %s, %w", c.Name, err)
```

**Comment by liornoy** [comment]
I wonder if we should be more explicit with the test description/title to remember that we're testing this use case of filtering addresses.
Maybe worth adding a small ginkgo.By to be more explicit?

File: e2etest/bgptests/bgp.go (hunk 1 above)
Comment (fedepaol) [reply e2etest/bgptests/bgp.go:1116]: No you are right, I am fixing it. This started simpler but didn't evolve in a readable way.

File: e2etest/bgptests/bgp.go (hunk 2 above)
Comment (fedepaol) [reply e2etest/bgptests/bgp.go:1118]: done

**Comment by liornoy** [comment]
lgtm

**Pull request merged**